docs/design/flow-vocabulary.md

```
@@ -0,0 +1,153 @@
+that sub-flow are prevented from sending until the sub-flow goes idle. This prevents values from the parent
+flow arriving while the sub-flow is busy processing.
+
+- **flow_blocks**: A map of `flow_id → set of function_ids`. When a function starts a job, it registers in
+  flow_blocks for its flow. When the flow goes idle, `remove_blocks` is called for each registered function,
+  resolving the output blocks that were holding back external senders.
+
+## Function State Transitions
+
+A function can be in one of five states: **Waiting**, **Ready**, **Blocked**, **Running**, **Completed**.
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)